### PR TITLE
Switch smart folder scanner to whitelist + add code file ingestion

### PR DESCRIPTION
**Pull request by shiba4life**
## Summary

- **Replace blacklist with whitelist**: Swap `BINARY_SKIP_EXTS` blacklist for an `INGESTIBLE_EXTS` whitelist covering data (`.json`, `.csv`, `.txt`, `.md`), code (`.js`, `.py`, `.rs`, `.go`, etc.), and config (`.yaml`, `.toml`, `.xml`) extensions. Only files matching the whitelist are collected during scanning.
- **Stop skipping coding projects**: Remove `is_coding_project()` and `PROJECT_MANIFEST_FILES` so `.md`, code, and config files inside coding project directories are now discoverable.
- **Add code metadata extraction**: New `extract_code_metadata()` function uses regex to pull function/class declarations and comments from source files. For `.js` files, tries Twitter export format first then falls back to code extraction. Config files are wrapped as text content.
- **Clean up dead code**: Remove always-empty `binary_skipped` variable, stale "binary-skipped" comments, and redundant `test_scan_does_not_skip_root_coding_project` test.

## Test plan

- [x] `cargo test --lib smart_folder_scanner` — 10 tests (whitelist validation)
- [x] `cargo test --lib file_conversion` — 20 tests (code metadata, JS fallback, config wrapping, edge cases)
- [x] `cargo test --lib smart_folder` — 47 tests (scanner + classifier + integration)
- [x] `cargo test --test twitter_export_scan_test -- test_twitter_export_scanner_excludes_media` — integration test
- [x] `cargo test --lib` — 448/448 pass
- [x] `cargo clippy --lib` — 0 warnings

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Adds ingest support for many document, spreadsheet, presentation, email and contact formats and improved detection of Twitter export files.

* **Refactor**
  * Non-JSON uploads now produce consistent, structured JSON-like metadata for more reliable downstream processing.

* **Tests**
  * Expanded and updated test coverage to validate broader file types, updated ingestible selection, and new parsing scenarios.

### CODE REVIEW

File: src/ingestion/smart_folder_scanner.rs [hunk 1]
```diff
@@ -207,31 +207,36 @@ pub fn compute_file_hash(file_path: &Path) -> IngestionResult<String> {
...
+/// Data file extensions (JSON, CSV, plain text, markdown).
+pub const DATA_EXTS: &[&str] = &["json", "csv", "txt", "md"];
+
+/// Document file extensions (personal data — handled by LLM classifier).
+pub const DOC_EXTS: &[&str] = &[
+    "pdf", "doc", "docx", "rtf", "odt", "pages",
+    "xls", "xlsx", "ods", "numbers",
+    "pptx", "ppt", "odp", "key",
+    "eml", "mbox", "vcf",
+];
```
[the remark this answers is not in the document]
Comment (shiba4life) [reply src/ingestion/smart_folder_scanner.rs:219]: This is a false positive. The ingestion pipeline already handles binary document formats via a dual-path architecture in routes.rs process_single_file_via_smart_folder:

1. read_file_with_hash() is tried first (handles text-based formats)
2. On failure, it falls back to convert_file_to_json() which uses the file_to_json crate with AI vision models for binary formats (PDF, DOCX, XLSX, images, etc.)

DOC_EXTS in the scanner whitelist is intentionally correct - these files need to be discovered during scanning and classified by the LLM. The actual conversion happens through the fallback path, not through parse_content_by_ext.
